From 93915676aa4751ce5a682e6b269b4ba15c14a873 Mon Sep 17 00:00:00 2001
From: Heinrich Schuchardt <xypron.glpk@gmx.de>
Date: Thu, 30 May 2019 11:18:11 +0200
Subject: [PATCH 2/3] fs: fat: Kconfig option for codepage

In the UEFI subsystem we depend on the codepage of the FAT filesystem
in the implementation of the Unicode collation protocol.

Add a choice between codepages 1250 (used by Windows) and 437 (typically
used by Linux).

Signed-off-by: Heinrich Schuchardt <xypron.glpk@gmx.de>
---
 fs/fat/Kconfig | 21 +++++++++++++++++++++
 1 file changed, 21 insertions(+)

diff --git a/fs/fat/Kconfig b/fs/fat/Kconfig
index 12524eeeed..4784583647 100644
--- a/fs/fat/Kconfig
+++ b/fs/fat/Kconfig
@@ -8,6 +8,27 @@ config FS_FAT
 
 if FS_FAT
 
+choice FAT_DEFAULT_CODEPAGE_CHOICE
+	prompt "Codepage used for file names"
+	default FAT_DEFAULT_CODEPAGE_437
+	help
+	  The fat filesystem can use different codepages to store filenames.
+	  Linux tends to use codepage 437. Windows uses codepage 1250. Other
+	  codepages currently are not supported.
+
+config FAT_DEFAULT_CODEPAGE_437
+	bool "Codepage 437"
+
+config FAT_DEFAULT_CODEPAGE_1250
+	bool "Codepage 1250"
+
+endchoice
+
+config FAT_DEFAULT_CODEPAGE
+	int
+	default 437 if FAT_DEFAULT_CODEPAGE_437
+	default 1250 if FAT_DEFAULT_CODEPAGE_1250
+
 config FAT_WRITE
 	bool "Enable FAT filesystem write support"
 	help
-- 
2.20.1

